solr/core/src/java/org/apache/solr/api/V2HttpCall.java

```diff
@@ -360,6 +366,80 @@ protected void execute(SolrQueryResponse rsp) {
     SolrCore.postDecorateResponse(handler, solrReq, rsp);
   }
 
+  @Override
+  protected void populateTracingSpan(Span span) {
+    // Set db.instance
+    String coreOrColName = this.origCorename;
+    if (coreOrColName == null) {
+      Map<String, String> pathTemplateValues = getUrlParts(); // == solrReq.getPathTemplateValues()
+      coreOrColName = pathTemplateValues.get("collection");
+      if (coreOrColName == null) {
+        coreOrColName = pathTemplateValues.get("core");
+      }
+    }
+    if (coreOrColName != null) {
+      span.setTag(Tags.DB_INSTANCE, coreOrColName);
+    }
+
+    // Get the templatize-ed path
+    String path;
+    if (api instanceof AnnotatedApi) {
+      // ideal scenario; eventually everything might be AnnotatedApi?
+      path = ((AnnotatedApi) api).getEndPoint().path()[0]; // consider first to be primary
+    } else {
+      path = computeEndpointPath();
+      // TODO consider getValidators, looking for command & path?
+    }
+
+    String verb = req.getMethod().toLowerCase(Locale.ROOT);
+    // if this api has commands ...
+    final Map<String, JsonSchemaValidator> validators = getValidators(); // should be cached
+    if (validators != null && validators.isEmpty() == false && solrReq != null) {
```

Comment:
This is somewhat of a hack for performance.  Ideally, solrReq.getCommands below would eagerly grab the validators to see none and then return an empty list before bothering with grabbing the streams.  I could do a separate PR for that. CC @noblepaul

Reply:
The most common usecase will not have any commands. So, avoid getting the validators and the verb is absent

Reply:
I just wanted this part of the code to be as cheap as posisble